rfc/rfc-78/rfc-78.md

```diff
@@ -0,0 +1,220 @@
+in 1.x, log files could be rolledback (deleted) by a concurrent rollback. So, the FSV should ensure it ignores the uncommitted log files. 
+
+
+### FileSystemView:
+- FSV building/reading should account for commit instants in both 0.x and 1.x formats. Since completion time log file ordering is not supported in 0.16.0 (that’s our current assumption), we may not need to support exactly how a FSV in 1.x reader supports. But below items should be supported.
+    - File slicing should be intact. For a single writer and OCC based writer in 1.x, the mapping of log files to base instant time or file slice should be same across 1.x reader and 0.16.0 reader.
+    - Log file ordering will follow 0.16.0 logic. I.e. log version followed by write token based comparator. Even if there are log files which completed in different order using 1.x writer, since we don’t plan to support that feature in 0.16.0 reader, we can try to maintain parity with 0.16.0 reader or log file ordering.
+    - We might have to revisit this if we plan to make NBCC default with MDT in 1.x
+- If not for completion time based read support, what will break, check [here](#Completion time based read in FileGroup/FileSlice initialization and log block reads) 
+
+### Table properties
+- payload type inference from payload class need to be ported.
+
+### MDT changes:
+- MDT schema changes need to be ported.
+
+### Log format changes:
+- New log header type need to be ported.
+- For unsupported features, meaningful errors should be thrown. For eg, if DV or partial update has been enabled and if those log files are read using 0.16.0 reader, we should fail and throw a meaningful error.
+
+### Incremental reads and time travel query: 
+- Incremental reads in 1.x is expected to have some changes and design is not fully out. So, until then we have to wait to design 0.16.x read support for 1.x tables for incremental queries. 
+- Time travel query: I don't see any changes in 1.x wrt time travel query. So, as per master, we are still using commit time(i.e not completed time) to serve time travel query. Until we make any changes to that, we do not need to add any additional support to 0.16.x reader. But if we plan to make changes in 1.x, we might have to revisit this. 
+
+## 0.16.0 ➝ 1.0 upgrade steps
+This will be an automatic upgrade for users when they start using 1.x hudi library. Listing the changes we might need to do with the upgrade.
+- Rewrite archival timeline to LSM based timeline.
+- Do not touch active timeline, since there could be concurrent readers reading the table. So, 1.x reader should be capable of handling timeline w/ a mix of 0.x commit files and 1.x commit files.
+- No changes to log reader.
+- Check custom payload class in table properties and switch to payload type.
+- Trigger compaction for latest file slices. We do not want a single file slice having a mix of log files from 0.x and log files from 1.x. So, we will trigger a full compaction 
```

Comment:
yes, we are thinking we should avoid full compaction in upgrade code path. downgrade should be fine (since anyways table might be in corrupt state or something). but upgrade lies in fast path and we could be having streaming pipelines. and w/ MDT(since its MOR), we can't afford to do full compaction. Lets review the RFC to gauge the complexity we are running into and take a call.